### ISSUE TITLE
Specify Gridline Alignment with Ticks

### ISSUE THREAD

**Issue by m4dc4p** [palantir/plottable#3075]
Gridlines can either line up with ticks on each axis (which makes sense for numeric axes & sparse plots), or they can be drawn in-between each tick (which makes sense for bar graphs). 

We should be able to specify which type of alignment we want. Proposal:
- A property named "alignment" which takes two values:
  - `center` - Gridlines are aligned with each tick.
  - `between` - Gridlines are drawn halfway between each tick.

Note that the alignment of gridlines can be different for X & Y axes.

Note, however, that alignment only seems to make sense for Category axis. I don't think you'd ever want to draw gridlines off-tick on a numeric axis.

Proposal: Add this property to the Scales.Category class; If no value is given, gridlines should draw on ticks (just like they are now).

**Comment by adidahiya** [comment]
Agreed that it should go on the `Scales.Category` class. Should the property name be a little more verbose? Perhaps `gridAlignment`?

**Comment by m4dc4p** [comment]
`gridAlignment` makes sense to me.

**Comment by adidahiya** [comment]
cool let's do it 👍

**Comment by adidahiya** [comment]
ping here @m4dc4p, any ETA on when you might be able to implement the discussed changes? no rush, just wondering.

**Comment by m4dc4p** [comment]
No worries. I hope to get a PR up by next Friday (5/13).

**Comment by m4dc4p** [comment]
@adidahiya: my circumstances have changed (new job) and I'm unlikely to do this work. Hopefully someone else can pick it up.

**Comment by adidahiya** [comment]
@m4dc4p ok, that's cool, thanks for the update. Will leave it up to anyone else interested in the feature.
